wasm/CMakeLists.txt

```diff
@@ -130,7 +130,8 @@ if(NOT EXISTS ${DEP_PREFIX}/lib/libz.a)
             && export CXX=\"${CMAKE_CXX_COMPILER_LAUNCHER} ${CMAKE_CXX_COMPILER}\"  \
             && export AR=${CMAKE_AR}                                                \
             && export RANLIB=${CMAKE_RANLIB}                                        \
-            && export CFLAGS=\"--sysroot=${CMAKE_SYSROOT}\"                         \
+            && export CFLAGS=\"--sysroot=${CMAKE_SYSROOT} --target=wasm32-wasi\"    \
+            && export LDFLAGS=\"-target wasm32-wasi\"                              \
```

Comment:
I was wrong about this being needed. Notice how the CFLAGS value gets passed to the linker